packages/cli/src/databases/repositories/test-case-execution.repository.ee.ts

```diff
@@ -0,0 +1,88 @@
+import { Service } from '@n8n/di';
+import { DataSource, In, Not, Repository } from '@n8n/typeorm';
+import type { DeepPartial } from '@n8n/typeorm/common/DeepPartial';
+
+import { TestCaseExecution } from '@/databases/entities/test-case-execution.ee';
+
+@Service()
+export class TestCaseExecutionRepository extends Repository<TestCaseExecution> {
+	constructor(dataSource: DataSource) {
+		super(TestCaseExecution, dataSource.manager);
+	}
+
+	async createBatch(testRunId: string, pastExecutionIds: string[]) {
+		const mappings = this.create(
+			pastExecutionIds.map<DeepPartial<TestCaseExecution>>((id) => ({
+				testRun: {
+					id: testRunId,
+				},
+				pastExecution: {
+					id,
+				},
+				status: 'new',
+			})),
+		);
+
+		return await this.save(mappings);
+	}
+
+	async markAsRunning(testRunId: string, pastExecutionId: string, executionId: string) {
+		return await this.update(
+			{ testRun: { id: testRunId }, pastExecutionId },
+			{
+				status: 'running',
+				executionId,
+				runAt: new Date(),
+			},
+		);
+	}
+
+	async markAsEvaluationRunning(
+		testRunId: string,
+		pastExecutionId: string,
+		evaluationExecutionId: string,
+	) {
+		return await this.update(
+			{ testRun: { id: testRunId }, pastExecutionId },
+			{
+				status: 'evaluation_running',
+				evaluationExecutionId,
+			},
+		);
+	}
+
+	async markAsCompleted(
+		testRunId: string,
+		pastExecutionId: string,
+		metrics: Record<string, number>,
```

Comment:
In the entity file the type is `Record<string, number | boolean>;`. Should it have the boolean or not?

Reply:
Nope, that's legacy, thanks for noticing!